src/TikTok/Provider.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace SocialiteProviders\TikTok;
+
+use Illuminate\Support\Arr;
+
+use SocialiteProviders\Manager\OAuth2\User;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+
+use Laravel\Socialite\Two\InvalidStateException;
+
+class Provider extends AbstractProvider
+{
+    /**
+     * Unique Provider Identifier.
+     */
+    public const IDENTIFIER = 'TIKTOK';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = [
+        "user.info.basic"
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return "https://open-api.tiktok.com/platform/oauth/connect?" . http_build_query([
+            "client_key" => $this->clientId,
+            "state" => $state,
+            "response_type" => "code",
+            "scope" => $this->formatScopes($this->getScopes(), $this->scopeSeparator),
+            "redirect_uri" => $this->redirectUrl,
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function user()
+    {
+        if($this->user) { return $this->user; }
+
+        if($this->hasInvalidState()) {
+            throw new InvalidStateException();
+        }
+
+        $response = $this->getAccessTokenResponse($this->getCode());
+
+        $token = Arr::get($response, "data.access_token");
+
+        $this->user = $this->mapUserToObject(
+            $this->getUserByToken([$token, Arr::get($response, "data.open_id")])
+        );
+
+        return $this->user->setToken($token)
+            ->setExpiresIn(Arr::get($response, "data.expires_in"))
+            ->setRefreshToken(Arr::get($response, "data.refresh_token"));
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getTokenUrl()
+    {
+        return "https://open-api.tiktok.com/oauth/access_token/";
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenFields($code)
+    {
+        $fields = [
+            "client_key" => $this->clientId,
+            "client_secret" => $this->clientSecret,
+            "code" => $code,
+            "grant_type" => "authorization_code",
+        ];
+
+        return $fields;
+    }
+
+    /**
+     * Get TikTok user by token.
+     * 
+     * @param array $data
+     * @return mixed
+     */
+    protected function getUserByToken($data)
+    {
+        // Note: The TikTok api does not currently have a way to get the user data
+        // with only the access token.
```

Comment:
Not sure what you mean here?

Reply:
The TikTok API does not have an endpoint where it returns the authorized user based on the access token only. You also have to give it the open-id to get the user information.

Thats why I've had to change $data to be an array where $data[0] is the access token and $data[1] is the user's open_id.